### PR TITLE
Skip empty chucks for BodyStream and SizedStream

### LINKED ISSUE #1267: Streaming response - inconsistent behaviour
I've ran into the following issue recently while returning chunked encoded response:

I stream some response by chunk and some chunks are empty. 
What happens is that if I request the endpoint requiring compression I'm getting the full response, but if I request it uncompressed I get only the part of data (the one before the first empty chunk).

It aligns with HTTP standard in a way: chunked response is terminated by a chunk with `0` length, but makes the usage of the framework a bit confusing (it took me a day to figure out the reason of this wired behavior).

My current suggestion would be that `actix-web` should just drop empty chunks from the given stream (this is what I did to solve the issue), but there might be another view on that.

## Code to reproduce

**Cargo.toml**
```
[package]
name = "actix-web-streaming-bug"
version = "0.1.0"
edition = "2018"

[dependencies]
actix-web="2.0.0"
actix-rt = "1.0.0"
bytes = "0.5.3"
futures = "0.3.1"
```

**main.rs**
```rust
use actix_web::{middleware, web, App, HttpRequest, HttpResponse, HttpServer, Responder};
use bytes::Bytes;

use futures::stream::iter;

use futures::{self, StreamExt};

async fn handle_request(_req: HttpRequest) -> impl Responder {
    let response = vec![
        Bytes::from("1\n"),
        Bytes::from(""),
        Bytes::from("2\n")
    ];
    HttpResponse::Ok().streaming(iter(response).map(|b| Ok(b) as Result<Bytes, ()>))
}

#[actix_rt::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(move || {
        App::new()
            .wrap(middleware::Compress::default())
            .service(web::resource("/").to(handle_request))
    })
    .bind("127.0.0.1:8080")
    .unwrap()
    .workers(1)
    .run()
    .await
}
```

```bash
$ curl 127.0.0.1:8080
1

$ curl --compressed 127.0.0.1:8080
1
2
```

### PR DESCRIPTION
**Pull request by tyranron**
Fixes #1267

This PR makes `BodyStream` and `SizedStream` in `actix-http` crate to skip empty chunks whicl streaming HTTP response, so omit unexpected termination.

### CODE REVIEW

**Comment by tyranron** [comment]
@JohnTitor I've implemented option 2 described [here](https://github.com/actix/actix-web/issues/1267#issuecomment-576777282) as the most preferred among commenters. But I'll change it if we should go with another one.

File: actix-http/src/body.rs [hunk 1]
```diff
@@ -589,4 +607,45 @@ mod tests {
...
+        async fn skips_empty_chunks() {
+            let body = SizedStream::new(
+                2,
+                stream::iter(
+                    ["1", "", "2"]
+                        .iter()
+                        .map(|&v| Ok(Bytes::from(v))),
+                ),
+            );
+            assert_eq!(read_all(body).await.unwrap(), Bytes::from("12"));
```
Comment (robjtede) [inline actix-http/src/body.rs]: These tests pass with the old code that doesnt filter empty chunks.

It's clearer and more explicit to write the tests for this as test_vec, using manual polling and output checking. Eg:

```rust
assert_eq!(
    poll_fn(|cx| body.poll_next(cx)).await.unwrap().ok(),
    Some(Bytes::from("1"))
);
assert_eq!(
    poll_fn(|cx| body.poll_next(cx)).await.unwrap().ok(),
    Some(Bytes::from("2"))
);
```

**Review by robjtede — CHANGES_REQUESTED** [review]

File: actix-http/src/body.rs (hunk 1 above)
Comment (tyranron) [reply actix-http/src/body.rs]: 🤦‍♂️

Thanks!

**Review by robjtede — APPROVED** [review]

**Review by cdbattags — APPROVED** [review]

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Could you also update the changelog with the PR number?

**Comment by JohnTitor** [comment]
Thanks!

**Pull request merged**